src/bartiq/visualisations.py

```diff
@@ -0,0 +1,191 @@
+from __future__ import annotations
+
+from numbers import Number
+from typing import Literal
+
+import pandas as pd
+import plotly.express as px
+from plotly.graph_objs._figure import Figure as PlotlyFig
+
+from bartiq import CompiledRoutine, Resource
+
+
+class TreeMap:
+    """Plot treemaps for different resources given a numeric compiled routine.
+
+    Args:
+        routine: The routine to generate treemaps for.
+
+    Raises:
+        ValueError: If any resource in the top level routine has a non-numeric value.
+    """
+
+    COLUMNS = ["Routine", "Parent", "Contribution"]
+
+    def __init__(self, routine: CompiledRoutine):
+        self.routine = routine
+        if any(not isinstance(x.value, Number) for x in self.routine.resources.values()):
+            raise ValueError(f"{self.__class__.__name__} only accepts numeric routines.")
```

Comment:
I think I wrote this test, but it doesn't work. Change to:

```suggestion
        if not all(isinstance(x.value, Number) for x in self.routine.resources_values().values():
            raise ValueError(f"{self.__class__.__name__} only accepts numeric routines.")
```

We will also need a test to make sure this exception is thrown!